torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4504,6 +4504,23 @@ def _tensor(shape, dtype=dtype, low=None, high=None):
 
     return [SampleInput(tensor, args=args) for tensor, args in test_cases]
 
+def sample_inputs_attn(op_info, device, dtype, requires_grad, **kwargs):
+    def _tensor(shape):
+        return make_tensor(shape, device, dtype, low=None, high=None, requires_grad=requires_grad)
+
+    test_dims = [1, 3, 7]
+
+    def generator():
+        for dim1 in test_dims:
+            for dim2 in test_dims:
+                for dim3 in test_dims:
```

Comment:
nit: `for dim1, dim2, dim3 in itertools.product(test_dims, test_dims, test_dims):` makes it so that we don't have to do three layers of indentation